frame/merkle-mountain-range/src/tests.rs

```diff
@@ -555,7 +572,8 @@ fn should_verify_batch_proofs() {
 		let leaves_set: Vec<Vec<u64>> = (0..=n).into_iter().powerset().skip(1).collect();
 
 		leaves_set.iter().for_each(|leaves_subset| {
-			generate_and_verify_batch_proof(&mut ext, leaves_subset, 0);
+			let block_numbers = leaf_indices_to_block_numbers(&mut ext, leaves_subset);
+			generate_and_verify_batch_proof(&mut ext, &block_numbers, 0);
```

Comment:
Can we do something like the following ?

```
	for n in 1..=10 {
		ext.execute_with(|| new_block());
		ext.persist_offchain_overlay();

		// generate powerset (skipping empty set) of all possible leaf combinations for mmr size n
		let blocks_set: Vec<Vec<u64>> = (1..=n).into_iter().powerset().skip(1).collect();

		blocks_set.iter().for_each(|blocks_subset| {
			generate_and_verify_batch_proof(&mut ext, &blocks_subset, 0);
			ext.persist_offchain_overlay();
		});
	}
```

And test with block numbers directly instead of converting leaf indices to block numbers. Also the same below and maybe we can remove the `leaf_indices_to_block_numbers()` function.